samples/SocialSample/Startup.cs

```diff
@@ -199,13 +206,10 @@ 127.0.0.1 MsSecSample.localhost.this
                         await context.Response.WriteAsync("<a href=\"?authscheme=" + type.AuthenticationScheme + "\">" + (type.Caption ?? "(suppressed)") + "</a><br>");
                     }
                     await context.Response.WriteAsync("</body></html>");
-                });
-            });
-
-            // Sign-out to remove the user cookie.
-            app.Map("/logout", signoutApp =>
-            {
-                signoutApp.Run(async context =>
+                    return;
+                }
+                
+                if (context.Request.Path.StartsWithSegments(new PathString("/logout")))
                 {
                     await context.Authentication.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
                     context.Response.ContentType = "text/html";
```

Comment:
Instead of fragile, I prefer the term powerful and immediate :)  But yeah the behavior is very different, I don't think we entirely know all the ramifications yet (i.e. map path issue)